base/path.jl

```diff
@@ -218,9 +218,9 @@ function joinpath(a::String, b::String)
     B, b = splitdrive(b)
     !isempty(B) && A != B && return string(B,b)
     C = isempty(B) ? A : B
-    isempty(a)                              ? string(C,b) :
-    contains(a[end:end], path_separator_re) ? string(C,a,b) :
-                                              string(C,a,pathsep(a,b),b)
+    isempty(a)                             ? string(C,b) :
+    isfound(path_separator_re, a[end:end]) ? string(C,a,b) :
```

Comment:
Not really related, but `a[end:end]` is a one-character string, right? So why call `isfound` rather than `==`? Because of the regex?